File: Medium/1759. Count Number of Homogenous Substrings/1759. Count Number of Homogenous Substrings.java
class Solution {
    public int countHomogenous(String s) {
        final int MOD = 1_000_000_007;
        int count = 0;
        int result = 0;

        for (int i = 0; i < s.length(); i++) {
            if (i == 0 || s.charAt(i) == s.charAt(i - 1)) {
                count++;
            } else {
                count = 1;
            }

            result = (result + count) % MOD;
        }

        return result;
    }
}

File: Medium/1759. Count Number of Homogenous Substrings/1759. Count Number of Homogenous Substrings.py
class Solution:
    def countHomogenous(self, s: str) -> int:
        MOD = 10**9 + 7
        count = 0
        result = 0

        for i in range(len(s)):
            if i == 0 or s[i] == s[i - 1]:
                count += 1
            else:
                count = 1

            result = (result + count) % MOD

        return result
